src/EnergyPlus/DataGlobalConstants.cc

```diff
@@ -652,6 +658,146 @@ namespace DataGlobalConstants {
+            FuelTypeNum = 5; // FuelTypeGasoline
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "FuelOilNo1")) {
+            FuelTypeNum = 6; // FuelTypeFuelOil1
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "FuelOilNo2")) {
+            FuelTypeNum = 7; // FuelTypeFuelOil2
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "OtherFuel1")) {
+            FuelTypeNum = 8; // FuelTypeOtherFuel1
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "OtherFuel2")) {
+            FuelTypeNum = 9; // FuelTypeOtherFuel2
+        } else {
+            FuelTypeErrorsFound = true;
+        }
+    }
+
+    void ValidateFuelTypeWithAssignResourceTypeNum(std::string FuelTypeInput)
+    {
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Dareum Nam
+        //       DATE WRITTEN   May 2020
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Validates fuel types and sets output strings with AssignResourceTypeNum(std::string const &ResourceTypeChar) (Boilers.cc and BoilerSteam.cc)
+
+        auto const SELECT_CASE_var(FuelTypeInput);
+
+        if (SELECT_CASE_var == "ELECTRICITY") {
+            FuelType = "Electric";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("ELECTRICITY");
+
+        } else if (SELECT_CASE_var == "NATURALGAS") {
+            FuelType = "Gas";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("NATURALGAS");
+
+        } else if (SELECT_CASE_var == "DIESEL") {
+            FuelType = "Diesel";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("DIESEL");
+
+        } else if (SELECT_CASE_var == "GASOLINE") {
+            FuelType = "Gasoline";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("GASOLINE");
+
+        } else if (SELECT_CASE_var == "COAL") {
+            FuelType = "Coal";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("COAL");
+
+        } else if (SELECT_CASE_var == "FUELOILNO1") {
+            FuelType = "FuelOil#1";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("FUELOIL#1");
+
+        } else if (SELECT_CASE_var == "FUELOILNO2") {
+            FuelType = "FuelOil#2";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("FUELOIL#2");
+
+        } else if (SELECT_CASE_var == "PROPANE") {
+            FuelType = "Propane";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("PROPANE");
+
+        } else if (SELECT_CASE_var == "OTHERFUEL1") {
+            FuelType = "OtherFuel1";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("OTHERFUEL1");
+
+        } else if (SELECT_CASE_var == "OTHERFUEL2") {
+            FuelType = "OtherFuel2";
+            FuelTypeNum = DataGlobalConstants::AssignResourceTypeNum("OTHERFUEL2");
+
+        } else {
+            FuelTypeErrorsFound = true;
```

Comment:
Instead of having a global variable for fuel type errors found, I would, for now:
- Make this function a `bool` instead of a `void`
- Return true if the function successfully matches a fuel type, and return false if not -- OR return true if errors were found and false if it succeeds, but just be clear about it.

Then the calling function can handle things appropriately.

Reply:
Be careful, if there are other errors in GetInput, then this function could over-ride ErrorsFound to false.

Reply:
I think this warrants another function argument for &ErrorsFound. Where an error is passed back only if true. @jmarrec updated this a while back. See GetFanInletNode as an example.